src/hatch/env/collectors/plugin/hooks.py

```diff
@@ -1,7 +1,9 @@
+from typing import Type
+
 from hatch.env.collectors.default import DefaultEnvironmentCollector
 from hatchling.plugin import hookimpl
 
 
 @hookimpl
-def hatch_register_environment_collector():
+def hatch_register_environment_collector() -> Type[DefaultEnvironmentCollector]:
```

Comment:
For the hooks I think we should instead express the type as the parent interface class